File: src/euler_008.jl
data = open("dat/euler_008.txt") do file
    readline(file)
end
data = parse(BigInt, data) |> digits

prod_of_13(x) = reduce(*, getindex(data, x:x+12))
function answer(data)
    map(prod_of_13, 1:length(data)-13) |> maximum
end

answer(data) |> print

File: src/euler_008.hs
import Data.Char

substring :: Int -> Int -> String -> String
substring start end text = take (end - start) (drop start text)

substrings13 :: String -> [String]
substrings13 string = [substring n (n+13) string | n <- [0..length string -13]]

prodString :: String -> Int
prodString string = foldl1 (*) (map (\x -> ord x - 48) string)

answer :: String -> Int
answer string = maximum $ map prodString (substrings13 string)

main = do
        contents <- readFile "dat/euler_008.txt"
        putStrLn $ show $ answer contents

